game-core/src/main/java/games/strategy/engine/framework/ui/DefaultGameChooserEntry.java

```diff
@@ -43,8 +44,6 @@ public void fullyParseGameData() throws GameParseException {
     // TODO: We should be setting this in the the constructor. At this point, you have to call
     // methods in the
     // correct order for things to work, and that is bads.
-    gameData = null;
-
```

Comment:
This is the core change

Reply:
@RoiEXLab Uh, any idea why this was setting it to null? Concerned there could be side effects of removing this.

Reply:
Well in case parsing fails, this null assignment replaces the shallow parsed GameData with null.
But I'd argue that this is a wrong behaviour because the way the class works is that it parses a shallow version in the constructor that throws an exception if that fails, so an object without corresponding GameData can't be constructed.
But I double checked all usages (and removed null checks) if that helps

Reply:
Alright. I'm good moving forward with this change then.